app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  def update_post_counter
+    user.increment!(:posts_counter)
+  end
```

Comment:
## Optional
- [ ] I suggest that you use one of the Rails Active Callbacks `after_save` to call the `update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private